### PR TITLE
Added lint and travis

### PR DESCRIPTION
**Pull request by cchan**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -39,6 +39,7 @@
     "gulp-sass-lint": "^1.1.1"
   },
   "dependencies": {
+    "eslint": "^2.11.1",
```
Comment (cchan) [inline package.json]: sindresorhus/eslint-config-xo-space#3, needs to be ^2.3.0 to work but currently gulp-eslint requires only ^2.0.0.

**Comment by lpan** [comment]
T R A V I S :panda_face:

File: package.json [hunk 2]
```diff
@@ -6,7 +6,9 @@
   "scripts": {
     "start": "gulp serve",
     "start:prod": "sudo systemctl start sexxi.xyz",
-    "build": "gulp build"
+    "build": "gulp build",
+    "pretest": "npm install -g gulp",
```
Comment (lpan) [inline package.json]: ~~What is `pretest` for :stuck_out_tongue: I am pretty sure `npm install` will install `gulp` locally~~

I assume its for travis

File: gulpfile.js [hunk 3]
```diff
@@ -19,21 +23,37 @@ gulp.task('serve', ['nodemon', 'sass', 'pug'], function() {
...
 
 gulp.task('build', ['sass', 'pug']);
-
 gulp.task('pug', function () {
   return gulp.src(PATHS.pug)
     .pipe(pug())
     .pipe(gulp.dest('dist'))
     .pipe(browserSync.stream());
 });
-
```
Comment (lpan) [inline gulpfile.js:29]: whyyyyyyy :cry: ???

File: gulpfile.js [hunk 4]
```diff
@@ -19,21 +23,37 @@ gulp.task('serve', ['nodemon', 'sass', 'pug'], function() {
...
+  return gulp.src(PATHS.eslint)
+    .pipe(eslint())
+    .pipe(eslint.format())
+    .pipe(eslint.failAfterError());
+});
+gulp.task('puglint', function(){
+  return gulp.src(PATHS.pug)
+    .pipe(puglint());
+});
+gulp.task('sasslint', function(){
```
Comment (l-pan) [inline gulpfile.js]: :cry: we prob should use another preset :stuck_out_tongue: this one is really not cool

File: gulpfile.js [hunk 5]
```diff
@@ -19,22 +23,38 @@ gulp.task('serve', ['nodemon', 'sass', 'pug'], function() {
...
 });
 
-gulp.task('nodemon', function () {
+gulp.task('lint', ['eslint', 'puglint', 'sasslint']);
+gulp.task('eslint', function() {
+  return gulp.src(PATHS.eslint)
+    .pipe(eslint({configFile: ".eslintrc.json"}))
+    .pipe(eslint.format())
+    .pipe(eslint.failAfterError());
+});
```
Comment (lpan) [inline gulpfile.js]: We should use another preset :stuck_out_tongue: the google one is a `smh`

**Comment by lpan** [comment]
Btw, we also can have something like `npm run ts` if you think its necessary :smile: 

``` json
ts: "npm run lint && npm test"
```

**Comment by cchan** [comment]
Due to all of our PRs being based on this PR now, I'm going to merge to master because this branch is effectively used as master anyway.

**Comment by cchan** [comment]
[gah I can't use gh lol]

**Pull request merged**